src/features/UpdatePowerShell.ts

```diff
@@ -116,39 +126,55 @@ export async function InvokePowerShellUpdateCheck(
     switch (result.id) {
         // Yes choice.
         case 0:
-            let script: string;
-            if (process.platform === "win32") {
+            if (isWindows) {
                 const msiMatcher = arch === "x86" ?
                     "win-x86.msi" : "win-x64.msi";
 
-                const assetUrl = release.assets.filter((asset: any) =>
-                    asset.name.indexOf(msiMatcher) >= 0)[0].browser_download_url;
-
-                // Grab MSI and run it.
-                // tslint:disable-next-line: max-line-length
-                script = `
-$randomFileName = [System.IO.Path]::GetRandomFileName()
-$tmpMsiPath = Microsoft.PowerShell.Management\\Join-Path ([System.IO.Path]::GetTempPath()) "$randomFileName.msi"
-Microsoft.PowerShell.Utility\\Invoke-RestMethod -Uri ${assetUrl} -OutFile $tmpMsiPath
-try
-{
-    Microsoft.PowerShell.Management\\Start-Process -Wait -Path $tmpMsiPath
-}
-finally
-{
-    Microsoft.PowerShell.Management\\Remove-Item $tmpMsiPath
-}`;
+                const asset = release.assets.filter((a: any) => a.name.indexOf(msiMatcher) >= 0)[0];
+                const msiDownloadPath = path.join(os.tmpdir(), asset.name);
+
+                const res = await fetch(asset.browser_download_url);
+                if (!res.ok) {
+                    throw new Error("unable to fetch MSI");
+                }
+
+                await window.withProgress({
+                    title: "Downloading PowerShell Installer...",
+                    location: ProgressLocation.Notification,
+                    cancellable: false,
+                },
+                async () => {
+                    // Streams the body of the request to a file.
+                    await streamPipeline(res.body, fs.createWriteStream(msiDownloadPath));
+                });
+
+                // Stop the Integrated Console session because Windows likes to hold on to files.
+                sessionManager.stop();
+
+                // Invoke the MSI via cmd.
+                const msi = spawn("cmd", [`/S /C ${msiDownloadPath}`], {
```

Comment:
Do we need to use `cmd`, or is there a more direct way of invoking the MSI (or is there a reason not to)?

Reply:
What about `"msiexec", [ "/i", msiDownloadPath ]`

Reply:
Yes! Changed to msiexec